feature/experimental/backup_nhg/ate_tests/backup_nhg_test/backup_nhg_test.go

```diff
@@ -329,96 +268,52 @@ func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
-	const (
-		// Prefix for recursive nexthop resolution.
-		recurPfx = "192.0.2.254/32"
-		// Next hop adjacency identifier that the recursive next hop group will point to.
-		recurNHID = 10000
-		// Recursive next hop address for dstPfx to forward to.
-		recurNH = "192.0.2.254"
-		// Next hop group adjacency identifier that the recurPfx will forward to.
-		recurNHGID = 103
-		// Next hop group ID that the dstPfx will forward to.
-		dstNHGID = 101
-		// Backup next hop group ID that the dstPfx will forward to.
-		dstBackupNHGID = 102
-
-		dutPort2ID, dutPort3ID = 10002, 10003
-	)
-
-	rnh := fluent.NextHopEntry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
-		WithIndex(recurNHID).WithIPAddress(recurNH)
-	nhg := fluent.NextHopGroupEntry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
-		WithID(dstNHGID).AddNextHop(recurNHID, 1).WithBackupNHG(dstBackupNHGID)
-	pfx := fluent.IPv4Entry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
-		WithPrefix(dstPfx).WithNextHopGroup(dstNHGID)
-
-	nh1 := fluent.NextHopEntry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
-		WithIndex(dutPort2ID).WithIPAddress(atePort2.IPv4)
-	rnhg := fluent.NextHopGroupEntry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
-		WithID(recurNHGID).AddNextHop(dutPort2ID, 1)
-	rpfx := fluent.IPv4Entry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
-		WithPrefix(recurPfx).WithNextHopGroup(recurNHGID)
-
-	nh2 := fluent.NextHopEntry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
-		WithIndex(dutPort3ID).WithIPAddress(atePort3.IPv4)
-	bnhg := fluent.NextHopGroupEntry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
-		WithID(dstBackupNHGID).AddNextHop(dutPort3ID, 1)
-
-	if del {
-		a.gribic.Modify().DeleteEntry(t, pfx, nhg, bnhg, rnhg, rnh, nh2, nh1)
-	} else {
-		a.gribic.Modify().AddEntry(t, nh1, nh2, rnh, rnhg, rpfx, bnhg, nhg, pfx)
-	}
-	if err := awaitTimeout(a.ctx, a.gribic, t, time.Minute); err != nil {
-		t.Fatalf("Await got error for entries: %v", err)
-	}
+// configureNetworkInstance configures vrf VRF-A and adds the vrf to port1, and configures vrf-B.
+func configureNetworkInstance(t *testing.T, dut *ondatra.DUTDevice) {
+	c := &oc.Root{}
+	ni := c.GetOrCreateNetworkInstance(vrfA)
+	ni.Description = ygot.String("Non Default routing instance VRF-A created for testing")
+	ni.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF
+	p1 := dut.Port(t, "port1")
+	niIntf := ni.GetOrCreateInterface(p1.Name())
+	niIntf.Subinterface = ygot.Uint32(0)
+	niIntf.Interface = ygot.String(p1.Name())
+	gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(vrfA).Config(), ni)
+
+	ni1 := c.GetOrCreateNetworkInstance(vrfB)
+	ni1.Description = ygot.String("Non Default routing instance VRF-B created for testing")
+	ni1.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF
+	gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(vrfB).Config(), ni1)
 }
```

Comment:
Can you please add the following deviation code to the end of configureNetworkInstance() function:
```
	if *deviations.ExplicitGRIBIUnderNetworkInstance {
		fptest.EnableGRIBIUnderNetworkInstance(t, dut, *deviations.DefaultNetworkInstance)
		fptest.EnableGRIBIUnderNetworkInstance(t, dut, vrfA)
		fptest.EnableGRIBIUnderNetworkInstance(t, dut, vrfB)
	}
```
thanks

Reply:
Addressed.